opentelemetry-sdk/src/trace/provider.rs

```diff
@@ -357,6 +395,7 @@ mod tests {
             })
             .build();
 
-        assert_service_name(no_service_name, None);
+        assert_resource(&no_service_name, SERVICE_NAME, None);
```

Comment:
nit  - as earlier this test is redundant, and can be removed. As we are checking the length in the next test.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/pull/1646/commits/7947daa611d98cdc6af89c10996936df865a1a39